src/utils/vc.js

```diff
@@ -295,3 +312,26 @@ export async function isVerifiedPresentation(presentation, challenge, domain, re
 export function buildDockCredentialStatus(registryId) {
   return { id: `${DockRevRegQualifier}${registryId}`, type: RevRegType };
 }
+
+/**
+ * The function uses `jsonschema` package to verify that the `credential`'s subject `credentialSubject` has the JSON
+ * schema `schema`
+ * @param {object} credential - The credential to use
+ * @param {object} schema - The schema to use
+ * @returns {Boolean} - Returns promise to an object or throws error
+ */
+export function validateCredentialSchema(credential, schema) {
+  const requiresID = schema.required && schema.required.indexOf('id') > -1;
+  const subjects = credential.credentialSubject.length ? credential.credentialSubject : [credential.credentialSubject];
```

Comment:
This will make `subjects` array of empty array if `credential.credentialSubject` is empty which is not what you want